apps/api/src/app/subscribers-v2/utils/create-subscriber.mapper.ts

```diff
@@ -0,0 +1,92 @@
+import { SubscriberCustomData, UserSessionData } from '@novu/shared';
+import { CreateOrUpdateSubscriberCommand } from '@novu/application-generic';
+import { IChannelCredentials, IChannelSettings, SubscriberEntity } from '@novu/dal';
+import { CreateSubscriberRequestDto } from '../dtos/create-subscriber.dto';
+import { ChannelSettingsDto, SubscriberResponseDto } from '../../subscribers/dtos';
+import { ChannelCredentials } from '../../shared/dtos/subscriber-channel';
+
+export function mapSubscriberRequestToCommand(
+  dto: CreateSubscriberRequestDto,
+  user: UserSessionData
+): CreateOrUpdateSubscriberCommand {
+  return {
+    organizationId: user.organizationId,
+    environmentId: user.environmentId,
+    subscriberId: dto.subscriberId,
+    email: dto.email ?? undefined,
+    firstName: dto.firstName ?? undefined,
+    lastName: dto.lastName ?? undefined,
+    phone: dto.phone ?? undefined,
+    avatar: dto.avatar ?? undefined,
+    locale: dto.locale ?? undefined,
+    data: dto.data ? mapCustomData(dto.data) : undefined,
+    timezone: dto.timezone,
+    isUpsert: false,
+  };
+}
+
+function isAllowedType(value: unknown): value is string | number | boolean | string[] | undefined {
+  return (
+    value === undefined ||
+    typeof value === 'string' ||
+    typeof value === 'number' ||
+    typeof value === 'boolean' ||
+    (Array.isArray(value) && value.every((item) => typeof item === 'string'))
+  );
+}
+
+function mapCustomData(data: Record<string, unknown>): SubscriberCustomData {
```

Comment:
@tatarco Why is this mapper necessary? Shouldn't we just copy-paste all the custom data as is? What if it's a nested data object?